### PR TITLE
Feat/ultralytics obb support

### PR DESCRIPTION
**Pull request by dimidagd**
- Add YoloOBB class adapter
- Use obb attribute from OBB ultralytics model outputs to create 51 polyline label.

## What changes are proposed in this pull request?

Model outputs from OBB ultralytics models were ignored. This commit enables use of *apply_model* method on a 51 dataset. 

```python
import fiftyone as fo
import fiftyone.zoo as foz
from ultralytics import YOLO

model = YOLO('yolov8n-obb.pt')
dataset = foz.load_zoo_dataset(
    'coco-2017',
    split='validation',
)
fo.apply_model(dataset, model)
```

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Introduced functionality to convert Ultralytics YOLO instance segmentations to FiftyOne format.
	- Added support for YOLO OBB detection models, including configuration and model handling enhancements.

### CODE REVIEW

**Review by jacobmarks — APPROVED** [review]
Overall this implementation looks solid to me, and it works! Noting that the OBB models from Ultralytics are trained on DOTA so don't really work on datasets like COCO :)

Awesome contribution @dimidagd 💪

If you're so inclined, also feel free to add these pretrained OBB models to the FiftyOne Model Zoo, so that downloading and location of checkpoint is managed by FiftyOne :) You can do so by adding them [here](https://github.com/voxel51/fiftyone/blob/develop/fiftyone/zoo/models/manifest-torch.json). In addition, you can add a section to the [Ultralytics integration docs](https://github.com/voxel51/fiftyone/blob/develop/docs/source/integrations/ultralytics.rst) inference section for OBB object detection. 

All that being said, if you don't have the time, we can also make sure these get in, so as to highlight the awesomeness of your work!!!
